pegen/pegen.c

```diff
@@ -503,3 +503,64 @@ seq_flatten(Parser *p, asdl_seq *seqs)
 
     return flattened_seq;
 }
+
+expr_ty
+seq_to_dotted_name(Parser *p, asdl_seq *seq)
+{
+    /* Create a string of the form a.b.c like Python/ast.c alias_for_dotted_name does
+       for dotted names */
+    if (asdl_seq_LEN(seq) == 1) {
+        return asdl_seq_GET(seq, 0);
+    }
+
+    // Get length of dotted string
+    ssize_t len = 0;
+    for (int i = 0, l = asdl_seq_LEN(seq); i < l; i++) {
+        expr_ty current_expr = asdl_seq_GET(seq, i);
+        ssize_t current_len = PyUnicode_GET_LENGTH(current_expr->v.Name.id);
+        len += current_len + 1;
+    }
+    --len; // Last name does not have a dot
+
+    PyObject *str = PyBytes_FromStringAndSize(NULL, len);
```

Comment:
I expect this won't be long enough if there are any non-ASCII characters in `seq`, which will cause the UTF8 encoding to take up more bytes than there are characters in the original.

Reply:
That's right. I wrongfully didn't think that through, thinking that it is not needed, since we are talking about modules and packages.

Reply:
@gvanrossum I still use `PyUnicode_GET_LENGTH` here, since `PyUnicode_GET_SIZE` is deprecated since 3.3 and I could not find something better. What would be the correct function to use here?